### PR TITLE
New route/get next katathon

### PR DESCRIPTION
**Pull request by AnnaVih**
The new route will return the closest event object.

### CODE REVIEW

File: server/controllers/index.js [hunk 1]
```diff
@@ -69,13 +72,15 @@ export const updateKata = async (req, res) => {
     const { name, slug, link, score } = req.body
 
     const katathon = await Katathon.findById(katathonId)
-    katathon.katas.map(kata => {
-      if(kata.kataId === kataId) {
+
+    katathon.katas = katathon.katas.map(kata => {
+      if(kata.kataId == kataId) {
         if(name) kata.name = name
```
Comment (webtechalex) [inline server/controllers/index.js]: Can we use `Object.assign()` here to merge available properties in to the kata object?
(You will need to create a new object from the `req.body` with the destructured properties first)
Then you can transform the `if` into a ternary:
`return (kata.kataId == kataId) ? Object.assign({}, kata, bodyCopy) : kata`

File: server/controllers/index.js [hunk 2]
```diff
@@ -151,8 +156,30 @@ export const updateKatathon = async (req, res) => {
 }
 
 
-export const nextKatathon = (req, res) => {
-  res.send('Next Katathon is coming soonddd')
+export const nextKatathon = async (req, res) => {
+  try {
+    const katathons = await Katathon.find({ completed: false }).sort({ date_created: -1 })
```
Comment (webtechalex) [inline server/controllers/index.js]: Why `dateCreated: -1`?

File: server/controllers/index.js [hunk 3]
```diff
@@ -151,8 +156,30 @@ export const updateKatathon = async (req, res) => {
...
+    if(katathons.length > 0) {
+      const nextKatathon = getNextEvent(katathons)
+      res.status(200).json({
+        result: 'Success',
+        data: nextKatathon,
+        message: 'Next Katathon found successfully'
+      })
+    }else {
+      res.status(404).json({
+        result: '404',
```
Comment (webtechalex) [inline server/controllers/index.js]: I think `result` should be: `Not Found` for a 404 status

File: server/controllers/index.js [hunk 4]
```diff
@@ -151,8 +156,30 @@ export const updateKatathon = async (req, res) => {
...
+        data: [],
+        message: 'No Katathon event found'
+      })
+    }
+  } catch (err) {
+    res.status(400).json({
+      result: 'failed',
```
Comment (webtechalex) [inline server/controllers/index.js]: Maybe overly picky, but capitalise `Failed` for consistency with `Success` and `Not Found`, suggested above.

**Review by webtechalex — CHANGES_REQUESTED** [review]

**Comment by AnnaVih** [comment]
@webtechalex Amended!)

**Review by webtechalex — APPROVED** [review]

**Pull request merged**